src/Agent.Plugins/PipelineArtifact/PipelineArtifactServer.cs

```diff
@@ -171,22 +171,40 @@ private BuildDropManager CreateBulidDropManager(AgentTaskPluginExecutionContext
             string[] minimatchFilters,
             CancellationToken cancellationToken)
         {
-            IDictionary<string, DedupIdentifier> artifactNameAndManifestId = new Dictionary<string, DedupIdentifier>();
-            foreach (var buildArtifact in buildArtifacts)
+            DownloadPipelineArtifactOptions options;
+            if (buildArtifacts.Count() > 1)
             {
+                IDictionary<string, DedupIdentifier> artifactNameAndManifestId = new Dictionary<string, DedupIdentifier>();
+                foreach (var buildArtifact in buildArtifacts)
+                {
+                    if (buildArtifact.Resource.Type != PipelineArtifactTypeName)
+                    {
+                        throw new ArgumentException("The artifact is not of the type Pipeline Artifact.");
+                    }
```

Comment:
Move this validation outside of both if/else 
like :
if (buildArtifacts.Any(ba => ba.Resource.Type != PipelineArtifactTypeName)) { throw ... }